### ISSUE TITLE
lastGrep should ignore ANSI escape sequences

### ISSUE THREAD

**Issue by szeiger** [sbt/sbt#4840]
sbt version: 1.2.8

`lastGrep` greps the raw output, including any ANSI color sequences, which leads to unexpected and unwanted effects, e.g.:

```
sbt:root> lastGrep error.* library/mimaReportBinaryIssues
[error]  * method foo()Int in class scala.collection.immutable.BitSet does not have a correspondent in other version
[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.immutable.BitSet.foo")
sbt:root> lastGrep \[.*error.* library/mimaReportBinaryIssues
[0m[error]  * method foo()Int in class scala.collection.immutable.BitSet does not have a correspondent in other version
[0m[error]    filter with: ProblemFilters.exclude[DirectMissingMethodProblem]("scala.collection.immutable.BitSet.foo")
sbt:root> lastGrep \[error.* library/mimaReportBinaryIssues
sbt:root>
```

`error` is red in this case, eveything else has the default color, so there are at least two escape sequences per line, one to switch to red and the other one to switch back.

It would be more useful to filter out all escape sequences and grep the filtered output.

Bonus points for keeping track of all ANSI attributes, mapping the positions back to the attributed soutce, and synthesizing a new escaped version.

**Comment by eed3si9n** [comment]
@szeiger Thanks for the report.

> It would be more useful to filter out all escape sequences and grep the filtered output.

I agree with your expectation.

**Comment by Eruis2579** [comment]
@eed3si9n , Can I work on this issue?

**Comment by eed3si9n** [comment]
@Eruis2579 Yes, and as per usual, check out our [Contributor's Guildeline](https://github.com/sbt/sbt/blob/develop/CONTRIBUTING.md) and:
1. Make sure you can reproduce the problem and the fix manually
2. Include tests for the change
3. Follow our AI assisted contributions guideline (please disclose AI usages in the PR description, if any)
4. Multiple contributors might work on this issue in parallel
5. Generally we recommend getting CI to pass on a forked repo before sending a pull request (pull requests that do not pass tests will be closed after a few days)
